### PR TITLE
index.md build from source improvements

### PR DESCRIPTION
**Pull request by danielrobbins**
Adding some more detail on building from source (latest version) vs. building a release. Also adding information on installing required libraries.

### CODE REVIEW

File: doc/index.md [hunk 1]
```diff
@@ -68,31 +68,53 @@ To run the testsuite, you'll also need:
...
 
 ```bash
 mkdir -p ~/go
 export GOPATH=~/go
+go get -d -v github.com/lxc/lxd/lxd
+cd $GOPATH/src/github.com/lxc/lxd
+make deps
+export CGO_CFLAGS="${CGO_CFLAGS} -I${GOPATH}/deps/sqlite/ -I${GOPATH}/deps/dqlite/include/ -I${GOPATH}/deps/raft/include/ -I${GOPATH}/deps/libco/"
+export CGO_LDFLAGS="${CGO_LDFLAGS} -L${GOPATH}/deps/sqlite/.libs/ -L${GOPATH}/deps/dqlite/.libs/ -L${GOPATH}/deps/raft/.libs -L${GOPATH}/deps/libco/"
+export LD_LIBRARY_PATH="${GOPATH}/deps/sqlite/.libs/:${GOPATH}/deps/dqlite/.libs/:${GOPATH}/deps/raft/.libs:${GOPATH}/deps/libco/:${LD_LIBRARY_PATH}"
```
Comment (stgraber) [inline doc/index.md]: I'd probably recommend putting a comment saying to follow what `make deps` tell you.
That would avoid this getting out of date as we change dependencies.

File: doc/index.md [hunk 2]
```diff
@@ -68,31 +68,53 @@ To run the testsuite, you'll also need:
...
 
 ```bash
-go get -d -v github.com/lxc/lxd/lxd
-cd $GOPATH/src/github.com/lxc/lxd
+cd lxd-3.18
+export GOPATH=$(pwd)/_dist
 make deps
 export CGO_CFLAGS="${CGO_CFLAGS} -I${GOPATH}/deps/sqlite/ -I${GOPATH}/deps/dqlite/include/ -I${GOPATH}/deps/raft/include/ -I${GOPATH}/deps/libco/"
 export CGO_LDFLAGS="${CGO_LDFLAGS} -L${GOPATH}/deps/sqlite/.libs/ -L${GOPATH}/deps/dqlite/.libs/ -L${GOPATH}/deps/raft/.libs -L${GOPATH}/deps/libco/"
 export LD_LIBRARY_PATH="${GOPATH}/deps/sqlite/.libs/:${GOPATH}/deps/dqlite/.libs/:${GOPATH}/deps/raft/.libs:${GOPATH}/deps/libco/:${LD_LIBRARY_PATH}"
```
Comment (stgraber) [inline doc/index.md]: Same comment as above, I think we should get rid of those in favor of telling the user to follow the instructions from `make deps`.

File: doc/index.md [hunk 3]
```diff
@@ -68,31 +68,53 @@ To run the testsuite, you'll also need:
...
+requisite libraries that will also need to be installed in order for LXD to function. These libraries can be installed
+using the following commands:
+
+```bash
+cd "${GOPATH}/deps/sqlite" && make DESTDIR=/ libdir=/usr/lib install
+cd "${GOPATH}/deps/raft" && make DESTDIR=/ libdir=/usr/lib install
+cd "${GOPATH}/deps/libco" && make DESTDIR=/ LIBDIR=/usr/lib install
+cd "${GOPATH}/deps/dqlite" && make DESTDIR=/ libdir=/usr/lib install
+```
+`DESTDIR` and/or `libdir`/`LIBDIR` can be adjusted as necessary to install the libraries to a suitable library path.
```
Comment (stgraber) [inline doc/index.md]: Might be worth mentioning in this section or in a separate one, that installing isn't necessary or even useful during development. Instead in such cases you just want to ensure that you have the variables from `make deps` set (usually through bashrc) and that you have `~/go/bin` in your PATH, at which point you can just run any of the binaries directly.

**Comment by stgraber** [comment]
Thanks! Left a few comments in-line. Also you'll need to make sure your commits include the `Signed-off-by` line before we can merge this.

**Comment by danielrobbins** [comment]
Thanks for the feedback. I wasn't familiar with the in-place running of the binaries. I will update the PR to reflect this and improve it.

**Comment by stgraber** [comment]
Thanks!

**Comment by danielrobbins** [comment]
OK, I made a bunch of improvements to the improvements and also rebased and added the Signed-off-by: line. Please review.

**Comment by stgraber** [comment]
Thanks!

**Pull request merged**